src/api/InfluxDB.cpp

```diff
@@ -0,0 +1,238 @@
+		print(log_debug, "api InfluxDB will use default database \"vzlogger\"", ch->name());
+		_database = "vzlogger";
+	} catch (vz::VZException &e) {
+		print(log_error, "api InfluxDB requires parameter \"database\" as string!", ch->name());
+		throw;
+	}
+
+try {
+		_measurement_name = optlist.lookup_string(pOptions, "measurement_name");
+		print(log_finest, "api InfluxDB using measurement name %s", ch->name(), _database.c_str());
+	} catch (vz::OptionNotFoundException &e) {
+		print(log_debug, "api InfluxDB will use default measurement name \"vzlogger\"", ch->name());
+		_measurement_name = "vzlogger";
+	} catch (vz::VZException &e) {
+		print(log_error, "api InfluxDB requires parameter \"measurement_name\" as string!", ch->name());
+		throw;
+	}
+
+try {
+		_curl_timeout = optlist.lookup_int(pOptions, "timeout");
+		print(log_finest, "api InfluxDB using curl timeout %i", ch->name(), _curl_timeout);
+	} catch (vz::OptionNotFoundException &e) {
+		_curl_timeout = 30;  //seconds
+		print(log_debug, "api InfluxDB will use default timeout %i", ch->name(), _curl_timeout);
+	} catch (vz::VZException &e) {
+		print(log_error, "api InfluxDB requires parameter \"timeout\" as int!", ch->name());
+		throw;
+	}
+
+try {
+		_max_batch_inserts = optlist.lookup_int(pOptions, "max_batch_inserts");
+		print(log_finest, "api InfluxDB using max batch inserts: %i", ch->name(), _max_batch_inserts);
+	} catch (vz::OptionNotFoundException &e) {
+		_max_batch_inserts = 4500;  //max lines per request
+		print(log_debug, "api InfluxDB will use default max_batch_inserts %i", ch->name(), _max_batch_inserts);
+	} catch (vz::VZException &e) {
+		print(log_error, "api InfluxDB requires parameter \"max_batch_inserts\" as int!", ch->name());
+		throw;
+	}
+
+	// build request url
+	_url = _host;
+	_url.append("/write");
+	_url.append("?db=");
+	_url.append(_database);
+	print(log_debug, "api InfluxDB using url %s", ch->name(), _url.c_str());
+
+}
+
+vz::api::InfluxDB::~InfluxDB() // destructor
+{
+}
+
+void vz::api::InfluxDB::send()
+{
+	long int http_code;
+	CURLcode curl_code;
+	std::string request_body;
+	Buffer::Ptr buf = channel()->buffer();
+	Buffer::iterator it;
+
+	_api.curl = curlSessionProvider ? curlSessionProvider->get_easy_session(_host + channel()->uuid()) : 0; // TODO add option to use parallel sessions. Simply add uuid() to the key.
+	if (!_api.curl) {
+		throw vz::VZException("CURL: cannot create handle.");
+	}
+
+	print(log_debug, "Buffer has %i items", channel()->name(), buf->size());
+
+	buf->lock();
+	int request_body_lines = 0;
+	for (it = buf->begin(); it != buf->end(); it++) {
+		if(request_body_lines >= _max_batch_inserts) {
+			print(log_debug, "reached maximum lines for InfluxDB insertion request.", channel()->name());
+			break;
+		}
+		print(log_finest, "Reading buffer: timestamp %lld value %f", channel()->name(), it->time_ms(), it->value());
+		request_body.append(_measurement_name);
+		request_body.append(",");
+		request_body.append("uuid=");
```

Comment:
append is quite a "expensive" function (might reallocate buffer, copy content,...). At least the "," and "uuid="... should be put into one append call. The compiler can't optimize this.